File: backend/src/main/java/com/bookbook/domain/report/dto/response/ReportDetailResponseDto.java
package com.bookbook.domain.report.dto.response;


import com.bookbook.domain.report.entity.Report;
import com.bookbook.domain.report.enums.ReportStatus;
import lombok.Builder;
import lombok.NonNull;

import java.time.LocalDateTime;

@Builder
public record ReportDetailResponseDto(
        @NonNull Long id,
        @NonNull ReportStatus status,
        @NonNull Long reporterUserId,
        @NonNull Long targetUserId,
        Long closerId,
        @NonNull String reason,
        @NonNull LocalDateTime createdDate,
        @NonNull LocalDateTime modifiedDate,
        @NonNull LocalDateTime reviewedDate
){
    public static ReportDetailResponseDto from(Report report){
        Long closerId = report.getCloser() == null
                ? null
                : report.getCloser().getId();

        return ReportDetailResponseDto.builder()
                .id(report.getId())
                .status(report.getStatus())
                .reporterUserId(report.getReporterUser().getId())
                .targetUserId(report.getTargetUser().getId())
                .closerId(closerId)
                .reason(report.getReason())
                .createdDate(report.getCreatedDate())
                .modifiedDate(report.getModifiedDate())
                .reviewedDate(report.getReviewedDate())
                .build();
    }


}

File: backend/src/main/java/com/bookbook/domain/report/dto/response/ReportDetailResponseDto.kt
package com.bookbook.domain.report.dto.response

import com.bookbook.domain.report.entity.Report
import com.bookbook.domain.report.enums.ReportStatus
import java.time.LocalDateTime

data class ReportDetailResponseDto(
    @JvmField
    val id: Long,
    val status: ReportStatus,
    val reporterUserId: Long,
    val targetUserId: Long,
    val closerId: Long?,
    val reason: String,
    val createdDate: LocalDateTime,
    val modifiedDate: LocalDateTime,
    val reviewedDate: LocalDateTime?
) {
    constructor(report: Report) : this(
        id = report.id ?: 0,
        status = report.status,
        reporterUserId = report.reporterUser.id ?: 0,
        targetUserId = report.targetUser.id ?: 0,
        closerId = report.closer?.id,
        reason = report.reason,
        createdDate = report.createdDate ?: LocalDateTime.MIN,
        modifiedDate = report.modifiedDate ?: LocalDateTime.MIN,
        reviewedDate = report.reviewedDate
    )
}